### ISSUE TITLE
Mobile connection keeps banning app on duckdns

### ISSUE THREAD

**Issue by Mariusthvdb** [home-assistant/iOS#1542]
**iOS device model, version and app version**

Model Name: iPhone 11 Pro
Software Version: 14.4.1
App version: 2021.3 (2021.77)

**Home Assistant Core Version**

2021.3.4
**Describe the bug**

Since 2 days, the iOS app does as above. I can always connect over Wifi, but when I disable the wifi on my iPhone, the iOS app gets banned on an IP address.(109.36.142.156), which is the mobile provider .

Not always, immediately after a restart, I can connect, only after a while this gets blocked and banned. Cant seem to find any logic in events for this to happen.

Using a browser window with the identical duckdns address works perfectly.
Has something changed in the app logic?

Ive also tried to delete the app and reinstall, but even that wont help, it immediately loads the old config, even though deleting the app warns one for also deleting all data.

second to this, it wont connect over a .local connection either, complaining my computer is not allowed, even while its is connected to the trusted networks most definitely.

**To Reproduce**

Fresh restart (deleted ip-bans.yaml). Fresh app install (have it find the duckdns instance while being on wifi). connect ok,. turnoff wifi. still ok (over 4g, connecting to duckdns instance). after a while, this simply gets disconnected showing the orange cogwheel bottom right. reloading the app, or reconnecting is allowed only twice (but unsuccessful) and then gives the not allowed, because ip-bans has kicked in.

**Expected behavior**

This is only happening since a day or 2, maybe since 2021.3.4(?) not sure. would love the 'old' behavior back. NO matter which, always connectable over 4G data and duckdns.

**Screenshots**

**Additional context**

**Comment by zacwest** [comment]
Do you get any kind of log or persistent notification for the ban? What is your ban config?

**Comment by Mariusthvdb** [comment]
yes I get the ban pers not.
```
http:
  ip_ban_enabled: true
  login_attempts_threshold: 10
```

**Comment by Mariusthvdb** [comment]
Guess I should be closing for now, this is too silly. 
Only thing I can think of is I configured the app with local network login, and on 4g data connection that isn't allowed?

**Comment by Mariusthvdb** [comment]
so Ive given this some time, and do see this banned list growing:
```
# Vodafone
109.36.142.158:
  banned_at: '2021-04-08T17:25:14.602011+00:00'

109.36.137.3:
  banned_at: '2021-04-12T15:31:39.088997+00:00'

109.36.132.154:
  banned_at: '2021-04-14T20:04:34.421523+00:00'

109.36.133.11:
  banned_at: '2021-04-16T12:09:06.555566+00:00'

109.36.136.3:
  banned_at: '2021-04-16T18:42:34.758758+00:00'

109.36.139.115:
  banned_at: '2021-04-19T15:52:01.934051+00:00'

109.36.143.229:
  banned_at: '2021-04-24T09:50:31.902356+00:00'

109.36.131.134:
  banned_at: '2021-04-27T10:17:38.561281+00:00'

109.36.132.65:
  banned_at: '2021-04-28T21:20:56.125708+00:00'

109.37.146.196:
  banned_at: '2021-05-05T12:13:59.771019+00:00'

109.37.146.128:
  banned_at: '2021-05-05T15:16:21.074881+00:00'
```

could it be the reason I posted above? 3 of the registered devices are out of my local network most of the time. If these were configured using local network (while at home of course), could these be treeing to call home and hence get banned?

**Comment by zacwest** [comment]
Can you share the User-Agent that us getting banned? If it is the frontend, I think the issue is on that side - there is something using an incorrect or expired auth token without the frontend prompting the app for a new one. I don't believe it will be affected by internal/external.

**Comment by Mariusthvdb** [comment]
sorry but no, the persistent notification doesn't mention a user-agent at all. could I lower a log level so debug for that to show up? if so, on which component?

**Comment by zacwest** [comment]
I believe the persistent notification says what User-Agent had a failed login attempt. Does that not appear?

**Comment by Mariusthvdb** [comment]
No, have a look , just came
In

![image](https://user-images.githubusercontent.com/33354141/117337008-25ab1680-ae9d-11eb-9d30-12e88d452276.png)

**Comment by zacwest** [comment]
I'm not sure which component does the IP banning in core, but it would be useful to see as much information about the source of the invalid login, since it can pinpoint between the app and the webview. I don't think the app-side logging is going to be too insightful here.

**Comment by TomBrien** [comment]
I believe those banned are generated by [`http`](https://www.home-assistant.io/integrations/http/).

**Comment by Mariusthvdb** [comment]
is that `homeassistant.components.websocket_api.http.connection` or `homeassistant.components.http`

**Comment by Mariusthvdb** [comment]
bingo:
```
2021-05-07 11:44:58 DEBUG (MainThread) [homeassistant.components.http.view] Serving /auth/token to 109.36.131.84 (auth: False)
2021-05-07 11:44:58 WARNING (MainThread) [homeassistant.components.http.ban] Login attempt or request with invalid authentication from 109.36.131.84 (109.36.131.84). (Home Assistant/2021.4.1 (io.robbie.HomeAssistant; build:2021.115; iOS 14.5.1) Alamofire/5.4.1)
```
and a bit later:
```
2021-05-07 11:49:58 DEBUG (MainThread) [homeassistant.components.http.view] Serving /api/websocket to 109.36.131.84 (auth: False)
2021-05-07 11:49:58 DEBUG (MainThread) [homeassistant.components.http.view] Serving /auth/token to 109.36.131.84 (auth: False)
2021-05-07 11:49:58 WARNING (MainThread) [homeassistant.components.http.ban] Login attempt or request with invalid authentication from 109.36.131.84 (109.36.131.84). (Home Assistant/2021.4.1 (io.robbie.HomeAssistant; build:2021.115; iOS 14.5.1) Alamofire/5.4.1)
````

![Uploading Schermafbeelding 2021-05-07 om 11.55.50.png…]()

**Comment by Mariusthvdb** [comment]
would you have any other suggestion on where and what to check, the list keeps growing. there's even a double ban ;-)

```
185.133.110.237:
  banned_at: '2021-05-11T18:23:12.513634+00:00'

185.133.110.237:
  banned_at: '2021-05-11T18:23:17.629284+00:00'
```

**Comment by zacwest** [comment]
I have not had a chance to look into this. I'm not sure what is causing it, or why it is having issue. That does imply it's coming from the iOS app and not the frontend, which means something weird is happening. One thing you can do is grab logs from the device just after it's IP banned: export log files in App Configuration and [share them to me](https://www.dropbox.com/request/CtVWqILBXZ41tmFSGK2y) -- I'll delete them after taking a look.

**Comment by Mariusthvdb** [comment]
> One thing you can do is grab logs from _the device_

thats just the thing isn't it? which of my devices is it? most  of the devices have the same app installed, (except for a few old models in the house) and I also have it as webpage in my Home Screen. Would be helpful if there was some identification in the error...

```
Login attempt or request with invalid authentication from 109.36.138.214 (109.36.138.214). (Home Assistant/2021.5 (io.robbie.HomeAssistant; build:2021.150; iOS 14.5.1) Alamofire/5.4.3)
Login attempt or request with invalid authentication from 109.36.138.214 (109.36.138.214). (Mozilla/5.0 (iPhone; CPU iPhone OS 14_5_1 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Home Assistant/2021.5 (io.robbie.HomeAssistant; build:2021.150; iOS 14.5.1) Mobile/HomeAssistant, like Safari)
```
What is the Alamofire/5.4.3 device referring to? is that a web browser on the phone? or maybe an older iOS device?

Killed a few more long lived tokens that weren't used for over 11 weeks, but that wont help probably, as the bans indicate recent usage...

**Comment by zacwest** [comment]
Alamofire is the networking library of the iOS app. Sounds like both it and the frontend are exhibiting this behavior.

**Comment by Mariusthvdb** [comment]
Since I am the one checking the insurance mostly, seems it would be my device. Or can this be caused by background traffic too, even when the app is not open.

**Comment by zacwest** [comment]
Almost certainly whatever happening here is happening in the background. Likely in the window of time where the app has an expired access token which needs refresh.

**Comment by Mariusthvdb** [comment]
hmm, still not much to go on I am afraid. Ive now logged out and re-logged in on my 3 most used devices (iOS app and webpage on my iPhone) and my Safari browser on the Mac) and see 3 new access tokens being created.

Let's see what happens next couple of hours.

**Comment by Mariusthvdb** [comment]
hmm, even got my one safari browser desktop to make a failed login attempt, and I can positively identify that because of the user agent 
```
2021-05-18 09:21:08 WARNING (MainThread) [homeassistant.components.http.ban] Login attempt or request with invalid authentication from router.asus.com (192.168.1.1). (Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_6) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/14.0.3 Safari/605.1.15)
```
this makes me wonder: why in the first place? could it have to do with the http settings I use?

```
http:
  ssl_certificate: /ssl/fullchain.pem
  ssl_key: /ssl/privkey.pem
  ip_ban_enabled: true
  login_attempts_threshold: 3
```

and 
```
homeassistant:
  name: Ha Rpi4
  internal_url: !secret external_url
#  internal_url: !secret internal_url
  external_url: !secret external_url
  latitude: !secret latitude
  longitude: !secret longitude
  elevation: !secret elevation
  unit_system: metric
  time_zone: !secret time_zone
  packages: !include_dir_named packages
  allowlist_external_dirs:
    - /config
  auth_providers:
   - type: homeassistant
   - type: trusted_networks
     trusted_networks:
#       - 127.0.0.1
       - 192.168.1.0/24
```

secondly, as I can identify the desktop browser by the myuseragent service, can we not do that in the app? would be nice to be able to positively identify the apps user agent to compare it to the one returned in the error in the log.

Maybe add it to the about: page?

**Comment by Mariusthvdb** [comment]
ok, so I finally got some more leads... changed the internal_url in the configuration.yaml to blank (had it set to external_url as you can see above). It immediately banned me on my iPhone when away from the local network, using the cellular network. Had not seen that for a while, so guess that has to do with? I do not  understand why it would do so, while the external url is set correctly though.

Found a log section on that time and dropped it in your link above, hope you can spot something noteworthy there. btw, the logs are in local time, while the bans are in Utc time? took me a second to realize that ;-)

**Comment by zacwest** [comment]
The logs you attached appear to be after the ban happened, any chance you have the time leading up to it? It would be useful to see what errors precipitated the ban.

**Comment by Mariusthvdb** [comment]
dropped the bit preceding to the ban. thanks for having such a detailed look!

btw, seeing this too, should I create a separate issue for that, or is it regular:

```
2021-05-17 22:41:01.156 [Info] [hakit-work-queue] [Environment.swift:53] init() > WebSocket: Sending: {"event_type":"state_changed","id":667,"type":"subscribe_events"}
2021-05-17 22:41:01.162 [Info] [hakit-work-queue] [Environment.swift:53] init() > WebSocket: Received: result success HARequestIdentifier(rawValue: 667)
2021-05-17 22:41:02.134 [Info] [hakit-work-queue] [Environment.swift:53] init() > WebSocket: Received: event: for HARequestIdentifier(rawValue: 667)
2021-05-17 22:41:02.134 [Info] [hakit-work-queue] [Environment.swift:53] init() > WebSocket: Received: event: for HARequestIdentifier(rawValue: 667)
2021-05-17 22:41:02.141 [Info] [hakit-work-queue] [Environment.swift:53] init() > WebSocket: Received: event: for HARequestIdentifier(rawValue: 667)
2021-05-17 22:41:03.294 [Info] [hakit-work-queue] [Environment.swift:53] init() > WebSocket: Received: event: for HARequestIdentifier(rawValue: 667)
2021-05-17 22:41:03.295 [Info] [hakit-work-queue] [Environment.swift:53] init() > WebSocket: Received: event: for HARequestIdentifier(rawValue: 667)
2021-05-17 22:41:03.409 [Info] [hakit-work-queue] [Environment.swift:53] init() > WebSocket: Received: event: for HARequestIdentifier(rawValue: 667)
2021-05-17 22:41:03.410 [Info] [hakit-work-queue] [Environment.swift:53] init() > WebSocket: Received: event: for HARequestIdentifier(rawValue: 667)
2021-05-17 22:41:05.207 [Info] [hakit-work-queue] [Environment.swift:53] init() > WebSocket: Received: event: for HARequestIdentifier(rawValue: 667)
2021-05-17 22:41:05.338 [Verbose] [main] [WebViewController.swift:747] handleHaptic(_:) > Handle haptic type light
2021-05-17 22:41:05.442 [Info] [hakit-work-queue] [Environment.swift:53] init() > WebSocket: Received: event: for HARequestIdentifier(rawValue: 667)
2021-05-17 22:41:05.442 [Info] [hakit-work-queue] [Environment.swift:53] init() > WebSocket: Received: event: for HARequestIdentifier(rawValue: 667)
2021-05-17 22:41:05.465 [Info] [hakit-work-queue] [Environment.swift:53] init() > WebSocket: Received: event: for HARequestIdentifier(rawValue: 667)
2021-05-17 22:41:05.469 [Info] [hakit-work-queue] [Environment.swift:53] init() > WebSocket: Received: event: for HARequestIdentifier(rawValue: 667)
2021-05-17 22:41:05.772 [Info] [hakit-work-queue] [Environment.swift:53] init() > WebSocket: Received: event: for HARequestIdentifier(rawValue: 667)
2021-05-17 22:41:05.772 [Info] [hakit-work-queue] [Environment.swift:53] init() > WebSocket: Received: event: for HARequestIdentifier(rawValue: 667)
2021-05-17 22:41:05.782 [Info] [hakit-work-queue] [Environment.swift:53] init() > WebSocket: Received: event: for HARequestIdentifier(rawValue: 667)
2021-05-17 22:41:05.898 [Info] [hakit-work-queue] [Environment.swift:53] init() > WebSocket: Received: event: for HARequestIdentifier(rawValue: 667)
2021-05-17 22:41:06.111 [Info] [hakit-work-queue] [Environment.swift:53] init() > WebSocket: Received: event: for HARequestIdentifier(rawValue: 667)
2021-05-17 22:41:06.112 [Info] [hakit-work-queue] [Environment.swift:53] init() > WebSocket: Received: event: for HARequestIdentifier(rawValue: 667)
2021-05-17 22:41:06.112 [Info] [hakit-work-queue] [Environment.swift:53] init() > WebSocket: Received: event: for HARequestIdentifier(rawValue: 667)
2021-05-17 22:41:06.137 [Info] [hakit-work-queue] [Environment.swift:53] init() > WebSocket: Received: event: for HARequestIdentifier(rawValue: 667)
2021-05-17 22:41:06.137 [Info] [hakit-work-queue] [Environment.swift:53] init() > WebSocket: Received: event: for HARequestIdentifier(rawValue: 667)
2021-05-17 22:41:06.368 [Info] [hakit-work-queue] [Environment.swift:53] init() > WebSocket: Received: event: for HARequestIdentifier(rawValue: 667)
2021-05-17 22:41:06.504 [Info] [hakit-work-queue] [Environment.swift:53] init() > WebSocket: Received: event: for HARequestIdentifier(rawValue: 667)
2021-05-17 22:41:06.516 [Info] [hakit-work-queue] [Environment.swift:53] init() > WebSocket: Received: event: for HARequestIdentifier(rawValue: 667)
2021-05-17 22:41:06.705 [Info] [hakit-work-queue] [Environment.swift:53] init() > WebSocket: Received: event: for HARequestIdentifier(rawValue: 667)
2021-05-17 22:41:06.732 [Info] [hakit-work-queue] [Environment.swift:53] init() > WebSocket: Received: event: for HARequestIdentifier(rawValue: 667)
2021-05-17 22:41:06.739 [Info] [hakit-work-queue] [Environment.swift:53] init() > WebSocket: Received: event: for HARequestIdentifier(rawValue: 667)
2021-05-17 22:41:06.830 [Info] [hakit-work-queue] [Environment.swift:53] init() > WebSocket: Received: event: for HARequestIdentifier(rawValue: 667)
2021-05-17 22:41:06.850 [Info] [hakit-work-queue] [Environment.swift:53] init() > WebSocket: Received: event: for HARequestIdentifier(rawValue: 667)
2021-05-17 22:41:06.866 [Info] [hakit-work-queue] [Environment.swift:53] init() > WebSocket: Received: event: for HARequestIdentifier(rawValue: 667)
2021-05-17 22:41:07.032 [Info] [hakit-work-queue] [Environment.swift:53] init() > WebSocket: Received: event: for HARequestIdentifier(rawValue: 667)
```

repeating huge amounts of lines, seems a bit overactive... and later on:
```
2021-05-18 09:23:13.699 [Info] [hakit-work-queue] [Environment.swift:53] init() > WebSocket: Received: result success HARequestIdentifier(rawValue: 699)
2021-05-18 09:23:13.993 [Info] [hakit-work-queue] [Environment.swift:53] init() > WebSocket: Sending: {"event_type":"state_changed","id":700,"type":"subscribe_events"}
2021-05-18 09:23:14.019 [Info] [hakit-work-queue] [Environment.swift:53] init() > WebSocket: Received: result success HARequestIdentifier(rawValue: 700)
2021-05-18 09:23:14.579 [Info] [hakit-work-queue] [Environment.swift:53] init() > WebSocket: Received: event: for HARequestIdentifier(rawValue: 700)
2021-05-18 09:23:14.762 [Info] [hakit-work-queue] [Environment.swift:53] init() > WebSocket: Received: event: for HARequestIdentifier(rawValue: 700)
2021-05-18 09:23:15.540 [Verbose] [main] [AccountRow.swift:150] fetchAvatar() > got user from user HAResponseCurrentUser(id: "3aredacted3", name: Optional("Marijn"), isOwner: true, isAdmin: true, credentials: [HAKit.HAResponseCurrentUser.Credential(type: "homeassistant", id: nil), HAKit.HAResponseCurrentUser.Credential(type: "trusted_networks", id: nil)], mfaModules: [HAKit.HAResponseCurrentUser.MFAModule(id: "totp", name: "Authenticator app", isEnabled: false)])
2021-05-18 09:23:15.541 [Error] [main] [AccountRow.swift:201] fetchAvatar() > failed to grab thumbnail: missingURL
2021-05-18 09:23:18.227 [Info] [hakit-work-queue] [Environment.swift:53] init() > WebSocket: Received: event: for HARequestIdentifier(rawValue: 700)
2021-05-18 09:23:18.227 [Info] [hakit-work-queue] [Environment.swift:53] init() > WebSocket: Received: event: for HARequestIdentifier(rawValue: 700)
2021-05-18 09:23:18.233 [Info] [hakit-work-queue] [Environment.swift:53] init() > WebSocket: Received: event: for HARequestIdentifier(rawValue: 700)
2021-05-18 09:23:18.234 [Info] [hakit-work-queue] [Environment.swift:53] init() > WebSocket: Received: event: for HARequestIdentifier(rawValue: 700)
2021-05-18 09:23:18.234 [Info] [hakit-work-queue] [Environment.swift:53] init() > WebSocket: Received: event: for HARequestIdentifier(rawValue: 700)
2021-05-18 09:23:18.235 [Error] [main] [AccountRow.swift:201] fetchAvatar() > failed to grab thumbnail: missingURL
2021-05-18 09:23:18.235 [Error] [main] [AccountRow.swift:201] fetchAvatar() > failed to grab thumbnail: missingURL
2021-05-18 09:23:18.245 [Info] [hakit-work-queue] [Environment.swift:53] init() > WebSocket: Received: event: for HARequestIdentifier(rawValue: 700)
2021-05-18 09:23:18.246 [Info] [hakit-work-queue] [Environment.swift:53] init() > WebSocket: Received: event: for HARequestIdentifier(rawValue: 700)
2021-05-18 09:23:18.247 [Error] [main] [AccountRow.swift:201] fetchAvatar() > failed to grab thumbnail: missingURL
2021-05-18 09:23:18.247 [Error] [main] [AccountRow.swift:201] fetchAvatar() > failed to grab thumbnail: missingURL
2021-05-18 09:23:18.247 [Error] [main] [AccountRow.swift:201] fetchAvatar() > failed to grab thumbnail: missingURL
2021-05-18 09:23:18.257 [Error] [main] [AccountRow.swift:201] fetchAvatar() > failed to grab thumbnail: missingURL
2021-05-18 09:23:18.257 [Error] [main] [AccountRow.swift:201] fetchAvatar() > failed to grab thumbnail: missingURL
```

**Comment by zacwest** [comment]
Those logs look normal, albeit full -- the WebSocket connection subscribes to all state changes, and that is happening a lot. It's not clear to me from the logs what's causing the issue still. It may be related to the WebSocket connection, which is new in 2021.4, but you've been having this issue longer.

That it is also occurring in Safari makes me wonder if it's something custom embedded in the page that isn't handling the token expiration/fetch cycle well; the frontend uses a unique auth path for the app, which requires querying the app using JavaScript and getting a token that way. It's possible there's an issue there.

You can identify the app via User-Agent -- it include `Home Assistant/2021.5` (for that version) in the User-Agent.

**Comment by Mariusthvdb** [comment]
> You can identify the app via User-Agent -- it include Home Assistant/2021.5 (for that version) in the User-Agent.

where is that in the app? I can find this in any of the settings pages?

About the Safari/App occurrences: Ive found now that the app is the more dominant cause for the bans. Using them side by side when on 4G, ive noticed that the safari browser can still see HA, and interact, while the App is already banned. I dont really understand that, because the ip is identical. 

also, this is now seen in the log ('error sending notification', and near  the bottom, 'login succes, reset') :
```
2021-05-20 09:47:24 ERROR (MainThread) [homeassistant.components.mobile_app.notify] Error sending notification to https://mobile-apps.home-assistant.io/api/sendPushNotification: ContentTypeError(RequestInfo(url=URL('https://mobile-apps.home-assistant.io/api/sendPushNotification'), method='POST', headers=<CIMultiDictProxy('Host': 'mobile-apps.home-assistant.io', 'User-Agent': 'HomeAssistant/2021.5.5 aiohttp/3.7.4.post0 Python/3.8', 'Accept': '*/*', 'Accept-Encoding': 'gzip, deflate', 'Content-Length': '537', 'Content-Type': 'application/json')>, real_url=URL('https://mobile-apps.home-assistant.io/api/sendPushNotification')), (), message='Attempt to decode JSON with unexpected mimetype: text/plain; charset=utf-8', headers=<CIMultiDictProxy('Connection': 'keep-alive', 'Content-Length': '14', 'Content-Type': 'text/plain; charset=utf-8', 'Accept-Ranges': 'bytes', 'Date': 'Thu, 20 May 2021 07:47:24 GMT', 'X-Served-By': 'cache-ams21048-AMS', 'X-Cache': 'MISS', 'X-Cache-Hits': '0', 'X-Timer': 'S1621496844.405332,VS0,VE174', 'Vary': 'x-fh-requested-host, accept-encoding')>)
2021-05-20 10:06:54 DEBUG (MainThread) [homeassistant.components.http.ban] Login success, reset failed login attempts counter from 109.36.131.219
2021-05-20 10:13:31 ERROR (MainThread) [homeassistant.components.mobile_app.notify] Error sending notification to https://mobile-apps.home-assistant.io/api/sendPushNotification: ContentTypeError(RequestInfo(url=URL('https://mobile-apps.home-assistant.io/api/sendPushNotification'), method='POST', headers=<CIMultiDictProxy('Host': 'mobile-apps.home-assistant.io', 'User-Agent': 'HomeAssistant/2021.5.5 aiohttp/3.7.4.post0 Python/3.8', 'Accept': '*/*', 'Accept-Encoding': 'gzip, deflate', 'Content-Length': '365', 'Content-Type': 'application/json')>, real_url=URL('https://mobile-apps.home-assistant.io/api/sendPushNotification')), (), message='Attempt to decode JSON with unexpected mimetype: text/plain; charset=utf-8', headers=<CIMultiDictProxy('Connection': 'keep-alive', 'Content-Length': '14', 'Content-Type': 'text/plain; charset=utf-8', 'Accept-Ranges': 'bytes', 'Date': 'Thu, 20 May 2021 08:13:31 GMT', 'X-Served-By': 'cache-ams21071-AMS', 'X-Cache': 'MISS', 'X-Cache-Hits': '0', 'X-Timer': 'S1621498411.497730,VS0,VE146', 'Vary': 'x-fh-requested-host, accept-encoding')>)
```

the reset however doesn't allow the App back in..

**Comment by zacwest** [comment]
I spent some time investigating another report of being banned, and I think `trusted_networks` is the cause. When you log into the app using the `trusted_network` login, that same authentication cannot be used off the trusted network. What happens is that every authenticated request to the server gets immediately considered to be invalid, so this has a delayed effect: it will only happen once the access token expires.

So the timeline looks something like this:
![image](https://user-images.githubusercontent.com/74188/121625188-d8840c80-ca27-11eb-8adf-527093d4edb0.png)

**Comment by Mariusthvdb** [comment]
Yeah I watched that on Discord.
Does this mean the fix makes it possible to logon with trusted_networks, and than go out (off the network) and still be able to connect without ip-ban? 
Or, is the fix that we should always use user/pw logins on devices that go out of the network regularly?

**Comment by Mariusthvdb** [comment]
I had tried everything....... and after having again explicitly set my app, and browser to HA using u/p versus trusted_networks, I still got this failed login attempt to 185.133.110.237... nothing to do with my devices at all, not with my provider, either mobile or Isp. really puzzling

**Comment by zacwest** [comment]
The fix is to get logged out rather than banned. If anything we learned from Discord that it is tough to be sure which login credential you are using. I suggest disabling trusted networks temporarily, restarting the server, resetting the app and then logging in. After that, enable trusted networks again. If this works (as it did for David) it is possible there is a login bug that is issuing the refresh token from a different provider.

**Comment by SkilledAlpaca** [comment]
This exact issue has popped up on my brand new HAOS install. I've had to disable IP ban section since it was locking all my mobile app devices out.

**Comment by DavidFW1960** [comment]
> This exact issue has popped up on my brand new HAOS install. I've had to disable IP ban section since it was locking all my mobile app devices out.

Make sure you are not authenticating using trusted networks.

**Comment by SkilledAlpaca** [comment]
> > This exact issue has popped up on my brand new HAOS install. I've had to disable IP ban section since it was locking all my mobile app devices out.
> 
> Make sure you are not authenticating using trusted networks.

I don't believe I was using trusted networks, unless trusted proxies cause the same issues. 

---

I'm unsure if it helps, but if anyone comes across this issue and wants one more thing to try, this _seems_ to have resolved my problems. Quick side note, I also use Cloudflare DNS and proxy my Home Assistant URL through CF. 

I commented out the `ip_ban_enabled` and `login_attempts_threshold` lines. Added `auth_providers` as `homeassistant` and rebooted the entire HAOS VM. I let this run for a couple days so each mobile device could check in and update it's auth methods (if that's even a thing). I then uncommented the `ip_ban_enabled` and `login_attempts_threshold` lines and restarted again. 

After restarting through the console, I had one failed login attempt, but this was due to my web session needing to be refreshed fully. Below is the relevant parts of my `configuration.yaml` file in case anyone needs it. 

```yaml
# Home Assistant
homeassistant:
  auth_mfa_modules:
    - type: totp
  auth_providers:      
    - type: homeassistant

# External Networking
http:
  use_x_forwarded_for: true
  trusted_proxies:
    - 192.168.1.5 # nginx-proxy-manager LXC
    - 173.245.48.0/20 # Cloudflare IP's - https://www.cloudflare.com/ips-v4
    - 103.21.244.0/22
    - 103.22.200.0/22
    - 103.31.4.0/22
    - 141.101.64.0/18
    - 108.162.192.0/18
    - 190.93.240.0/20
    - 188.114.96.0/20
    - 197.234.240.0/22
    - 198.41.128.0/17
    - 162.158.0.0/15
    - 104.16.0.0/13
    - 104.24.0.0/14
    - 172.64.0.0/13
    - 131.0.72.0/22 # End of Cloudflrare IP's
  ip_ban_enabled: true
  login_attempts_threshold: 5
```

---

Apologies for necro-posting on this issue as well, it was just the most relevant issue to my problems.
